Add parsing color_t (#2015)
f3d-app/f3d: doc/user/PARSING.md

## Color

The following formats are supported when parsing a color, case insensitive:

- R,G,B where R, G, B are doubles between 0 and 1
- #RRGGBB where RR, GG, BB are hexadecimal values
- hsl(H, S%, L%) where H \<= 360, S and L in percent
- hsv(H, S%, V%) where H \<= 360, S and V in percent
- hwb(H, W%, B%) where H \<= 360, W and B in percent

> - cmyk(C%, M%, Y%, K%) where C, M, Y, K in percent

Reviewer: add links to the w3schools stuff ?

Author: only hsv is not in w3school others ok

Reviewer: We are discussing it on discord right now

Reviewer: https://www.w3.org/TR/css-color-3/#rgb-color this ones looks better

> - color name, see [HTML color names](https://en.wikipedia.org/wiki/Web_colors#HTML_color_names)

Reviewer: should this link be used ?

https://htmlpreview.github.io/?https://github.com/Kitware/vtk-examples/blob/gh-pages/VTKNamedColorPatches.html

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Reviewer 2: yeah format as `#rrggbb` if all components are `<=1` and multiples of `1./255` makes sense to me, otherwise format as double vector

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Reviewer: @VittorioSanchez do you think you could do that in this PR or in another PR ?

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Author: So output in #rrggbb for hex and rgb() and double vector for others right ?

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Author: If so I think I can do it in this PR

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Reviewer: nice

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Reviewer: so this needs updating :)

> When formatting a color into a string, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Author: right ^^'

## Color

The following formats are supported when parsing a color, case insensitive:

> - R,G,B where R, G, B are doubles [0, 1]

Reviewer: ```suggestion
- R,G,B where R, G, B are doubles >= 0
```

> - #RRGGBB where RR, GG, BB are hexadecimal values

Reviewer: missing `#RGB`

> - rgb(R, G, B) where R, G and B \<= 255

Reviewer: ```suggestion
- rgb(R, G, B) where R, G and B are integer [0, 255]
```

Use the same phrasing everywhere below

- hsl(H, S%, L%) where H is integer [0, 360], S and L are integer [0, 100]
- hsv(H, S%, V%) where H is integer [0, 360], S and V are integer [0, 100]
- hwb(H, W%, B%) where H is integer [0, 360], W and B are integer [0, 100]
- cmyk(C%, M%, Y%, K%) where C, M, Y, K are integer [0, 100]
- [color name](https://htmlpreview.github.io/?https://github.com/Kitware/vtk-examples/blob/gh-pages/VTKNamedColorPatches.html)

See [W3C](https://www.w3.org/TR/css-color-3/#rgb-color) doc for more details on these formats.

> When formatting a color into a string, it is formatted as `#RRGGBB` if values are multiple of 255. Otherwise, it is formatted as `R,G,B` where R, G, B are doubles between 0 and 1.

Reviewer: ```suggestion
When formatting a color into a string, it is formatted as `#RRGGBB` if values are multiple of 255. Otherwise, it is formatted as vector of doubles.
```
